summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```diff
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Comment:
So, the code is actually wrong if we don't do it. `b.next` will not be bigger than `b` if `b.id == Long.MaxValue`. And I don't want to have it be silently wrong.

If you prefer we can throw an error, I guess, which I don't like either.

Honestly, `earliestTimeOf(b)` can't be a total function, which is another correctness issues. It probably needs to return `Option[Timestamp]`, but that's another issue.